### ISSUE TITLE
Mistake in pseudocode of paper?

### ISSUE THREAD

**Issue by horacejlee** [frank-xwang/debiased-pseudo-labeling#2]
Hi, I think there might be a mistake in the pseudocode in Algorithm 1 of the [paper](https://arxiv.org/pdf/2201.01490.pdf). In line 18 of the pseudocode, it says:
`p_hat = momentum * p_hat + (1 - momentum) * p_uw.detach().mean(dim=0)`

But `p_uw` refers to the _debiased_ pseudo-labels rather than the probability distribution from the classifier, because in line 12 it says:
`p_uw = F.softmax(l_uw - tau * torch.log(p_hat), dim=1))`.

So I think line 18 should instead be something like this (so that it matches your code [here](https://github.com/frank-xwang/debiased-pseudo-labeling/blob/ef5ca109e44bc13ac6a53e128e85af9ecad09ca9/main_DebiasPL.py#L602)):
`p_hat = momentum * p_hat + (1 - momentum) * F.softmax(l_uw, dim=-1).detach().mean(dim=0)`

**Comment by frank-xwang** [comment]
Hi, thanks for pointing this out. The typo will be corrected in the updated arXiv paper.
